### PR TITLE
Update Model for DG

### PR DESCRIPTION
**Pull request by Blizzeracz**

### CODE REVIEW

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 1]
```diff
@@ -4,18 +4,20 @@ skinparam arrowThickness 1.1
 skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
-AddressBook *-right-> "1" UniquePersonList
-AddressBook *-right-> "1" UniqueTagList
-UniqueTagList -[hidden]down- UniquePersonList
-UniqueTagList -[hidden]down- UniquePersonList
+AddressBook *-right-> "1" UniqueAnimalList
+AddressBook *-right-> "1" UniquePetIdList
```
Comment (euchangxian) [inline docs/diagrams/BetterModelClassDiagram.puml:8]: Should be a mistake here; Do fix it

**Review by euchangxian — CHANGES_REQUESTED** [review]

File: docs/diagrams/BetterModelClassDiagram.puml (hunk 1 above)
Comment (Blizzeracz) [reply docs/diagrams/BetterModelClassDiagram.puml:8]: Yep, Thank you

File: docs/diagrams/ModelClassDiagram.puml [hunk 2]
```diff
@@ -4,51 +4,51 @@ skinparam arrowThickness 1.1
...
-
-Class UniquePersonList
-Class Person
-Class Address
-Class Email
+Package AnimalModel as AnimalModelPackage <<Rectangle>>{
+Class "<<interface>>\nAnimalReadOnlyCatalog" as AnimalReadOnlyCatalog
+Class "<<interface>>\nAnimalReadOnlyAnimalUserPrefs" as AnimalReadOnlyAnimalUserPrefs
+Class "<<interface>>\nAnimalModel" as AnimalModel
+Class Catalog
```
[the remark this answers is not in the document]
Comment (Blizzeracz) [reply docs/diagrams/ModelClassDiagram.puml:11]: Yep thats right

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -122,21 +122,22 @@ How the parsing works:
...
-* stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a `ReadOnlyUserPref` objects.
-* does not depend on any of the other three components (as the `Model` represents data entities of the domain, they should make sense on their own without depending on other components)
 
 <box type="info" seamless>
 
-**Note:** An alternative (arguably, a more OOP) model is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique tag, instead of each `Person` needing their own `Tag` objects.<br>
+**Note:** An alternative (arguably, a more OOP) model is given below. It has a `PetId` list in the `Catalog`, which `Animal` can reference. This allows `Catalog` to only require `PetId` object, instead of each `Person` needing their own `PetId` objects.<br>
```
[the remark this answers is not in the document]
Comment (Blizzeracz) [reply docs/DeveloperGuide.md:140]: It should be better to reflect the class name instead of using them interchangeably. ThankS!

**Review by euchangxian — APPROVED** [review]
LGTM. Thanks for making the requested changes!

**Pull request merged**